dace/frontend/fortran/ast_transforms.py

```diff
@@ -958,12 +2260,1923 @@ def visit_Execution_Part_Node(self, node: ast_internal_classes.Execution_Part_No
+
+            func_arg = self.scope_vars.get_var(node.parent, node.arg.name)
+            node.type = func_arg.type
+            node.arg.type = func_arg.type
+            node.sizes = self._get_sizes(func_arg)
+            node.arg.sizes = self._get_sizes(func_arg)
+
+        else:
+            node.type = func_arg_name_type
+            node.sizes = self._get_sizes(node.arg)
+
+        return node
+
+    def visit_UnOp_Node(self, node: ast_internal_classes.UnOp_Node):
+        node.lval = self.visit(node.lval)
+        if node.lval.type != 'VOID':
+            node.type = node.lval.type
+            node.sizes = self._get_sizes(node.lval)
+            node.offsets = self._get_offsets(node.lval)
+        return node
+
+    def visit_Call_Expr_Node(self, node: ast_internal_classes.Call_Expr_Node):
+
+        from dace.frontend.fortran.intrinsics import FortranIntrinsics
+
+        new_args = []
+        for arg in node.args:
+            new_args.append(self.visit(arg))
+        node.args = new_args
+
+        sizes, offsets, return_type = FortranIntrinsics.output_size(node)
+        if sizes is not None:
+            node.sizes = sizes
+            node.offsets = offsets
+        else:
+            node.sizes = None
+            node.offsets = None
+
+        if return_type != 'VOID':
+            node.type = return_type
+
+        return node
+
+    def _get_type(self, node):
+
+        if isinstance(node, ast_internal_classes.Int_Literal_Node):
+            return 'INTEGER'
+        elif isinstance(node, ast_internal_classes.Real_Literal_Node):
+            return 'REAL'
+        elif isinstance(node, ast_internal_classes.Bool_Literal_Node):
+            return 'LOGICAL'
+        else:
+            return node.type
+
+    def visit_Int_Literal_Node(self, node: ast_internal_classes.Int_Literal_Node):
+        node.sizes = []
+        node.offsets = [1]
+        return node
+
+    def visit_Double_Literal_Node(self, node: ast_internal_classes.Double_Literal_Node):
+        node.sizes = []
+        node.offsets = [1]
+        return node
+
+    def visit_Real_Literal_Node(self, node: ast_internal_classes.Real_Literal_Node):
+        node.sizes = []
+        node.offsets = [1]
+        return node
+
+    def visit_Bool_Literal_Node(self, node: ast_internal_classes.Bool_Literal_Node):
+        node.sizes = []
+        node.offsets = [1]
+        return node
+
+    def _get_offsets(self, node):
+
+        if isinstance(node, (ast_internal_classes.Int_Literal_Node, ast_internal_classes.Real_Literal_Node,
+                             ast_internal_classes.Bool_Literal_Node)):
+            node.offsets = [1]
+        return node.offsets
+
+    def _get_sizes(self, node):
+
+        if isinstance(node, (ast_internal_classes.Int_Literal_Node, ast_internal_classes.Real_Literal_Node,
+                             ast_internal_classes.Bool_Literal_Node)):
+            node.sizes = []
+        return node.sizes
+
+
+class PointerRemoval(NodeTransformer):
```

Comment:
docstrings please! I have no idea what kind of POINTER is removed and in which cases